### ISSUE TITLE
[WD 0.3.4] Crash when entering ender

### ISSUE THREAD

**Issue by No3371** [Jackiecrazy/WarDance#21]
This is a interesting one, according to the stacktrace, it's a shulker attacking, and the sneaking damage calculation went wrong.

I feel like it's becuase the server is too busy or not finished reading chunks from another world.

However, I have not observed this happening to Nether.

```
java.lang.Error: ServerHangWatchdog detected that a single server tick took 60.00 seconds (should be max 0.05)
	at java.util.Comparator.lambda$thenComparing$36697e65$1(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.Comparator$$Lambda$17646/401640331.compare(Unknown Source) ~[?:?] {}
	at java.util.TreeMap.put(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.TreeSet.add(Unknown Source) ~[?:1.8.0_261] {}
	at net.minecraft.world.server.ServerTickList.func_219504_a(SourceFile:201) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerTickList.func_205362_a(SourceFile:194) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.ITickList.func_205360_a(SourceFile:9) ~[?:?] {re:classloading}
	at net.minecraft.world.chunk.ChunkPrimerTickList.func_205381_a(SourceFile:45) ~[?:?] {re:computing_frames,re:classloading}
	at net.minecraft.world.chunk.Chunk.func_222879_B(Chunk.java:762) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.world.server.ChunkManager.func_222955_c_(ChunkManager.java:648) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.world.server.ChunkManager$$Lambda$23181/460078934.apply(Unknown Source) ~[?:?] {}
	at com.mojang.datafixers.util.Either.lambda$mapLeft$0(Either.java:162) ~[?:?] {re:classloading,re:mixin}
	at com.mojang.datafixers.util.Either$$Lambda$22986/2618310.apply(Unknown Source) ~[?:?] {}
	at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[?:?] {re:classloading}
	at com.mojang.datafixers.util.Either.mapLeft(Either.java:162) ~[?:?] {re:classloading,re:mixin}
	at net.minecraft.world.server.ChunkManager.func_222976_a_(ChunkManager.java:646) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.world.server.ChunkManager$$Lambda$22821/1811500255.apply(Unknown Source) ~[?:?] {}
	at java.util.concurrent.CompletableFuture.uniApply(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.concurrent.CompletableFuture$Completion.run(Unknown Source) ~[?:1.8.0_261] {}
	at net.minecraft.world.chunk.ChunkTaskPriorityQueueSorter.func_219083_b(SourceFile:58) ~[?:?] {re:classloading}
	at net.minecraft.world.chunk.ChunkTaskPriorityQueueSorter$$Lambda$22862/1752670515.run(Unknown Source) ~[?:?] {}
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider$ChunkExecutor.func_213166_h(ServerChunkProvider.java:514) ~[?:?] {re:classloading}
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider$ChunkExecutor.func_213168_p(ServerChunkProvider.java:522) ~[?:?] {re:classloading}
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:127) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider.func_212849_a_(ServerChunkProvider.java:130) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.world.World.func_217353_a(World.java:167) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:observerlib:coremodmethod,pl:mixin:APP:sereneseasons.mixins.json:MixinWorld,pl:mixin:APP:blockshifter.mixins.json:WorldMixin,pl:mixin:APP:brazier-common.mixins.json:WorldMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.world.IWorldReader.func_217348_a(IWorldReader.java:112) ~[?:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.world.World.func_212866_a_(World.java:163) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:observerlib:coremodmethod,pl:mixin:APP:sereneseasons.mixins.json:MixinWorld,pl:mixin:APP:blockshifter.mixins.json:WorldMixin,pl:mixin:APP:brazier-common.mixins.json:WorldMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.world.World.func_180495_p(World.java:379) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:observerlib:coremodmethod,pl:mixin:APP:sereneseasons.mixins.json:MixinWorld,pl:mixin:APP:blockshifter.mixins.json:WorldMixin,pl:mixin:APP:brazier-common.mixins.json:WorldMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.world.IBlockReader.func_217297_a(IBlockReader.java:46) ~[?:?] {re:mixin,re:classloading}
	at net.minecraft.world.IBlockReader$$Lambda$24564/621933359.apply(Unknown Source) ~[?:?] {}
	at net.minecraft.world.IBlockReader.func_217300_a(IBlockReader.java:142) ~[?:?] {re:mixin,re:classloading}
	at net.minecraft.world.IBlockReader.func_217299_a(IBlockReader.java:45) ~[?:?] {re:mixin,re:classloading}
	at net.minecraft.entity.LivingEntity.func_70685_l(LivingEntity.java:2620) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:apotheosis:apothshieldblock,xf:fml:apotheosis:apothpotiondmg,pl:mixin:APP:armorcurve.mixins.json:ValueUpdateMixin,pl:mixin:APP:supplementaries.mixins.json:LivingEntityMixin,pl:mixin:APP:wardance.mixins.json:MixinAttackSpeed,pl:mixin:APP:spiderstpo.mixins.json:LivingEntityMixin,pl:mixin:APP:assets/shetiphiancore/shetiphiancore.mixins.json:MixinLivingEntity,pl:mixin:APP:META-INF/airhop.mixins.json:accessor.ILivingEntityAccessor,pl:mixin:A,pl:runtimedistcleaner:A}


	at jackiecrazy.wardance.handlers.EntityHandler.sneak(EntityHandler.java:151) ~[wardance:0.3.4] {re:mixin,re:classloading}


	at net.minecraftforge.eventbus.ASMEventHandler_299_EntityHandler_sneak_LivingVisibilityEvent.invoke(.dynamic) ~[?:?] {}
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85) ~[eventbus-4.0.0.jar:?] {}
	at net.minecraftforge.eventbus.EventBus$$Lambda$2572/2048932599.invoke(Unknown Source) ~[?:?] {}
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-4.0.0.jar:?] {}
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-4.0.0.jar:?] {}
	at net.minecraftforge.common.ForgeHooks.getEntityVisibilityMultiplier(ForgeHooks.java:420) ~[forge:?] {re:classloading,re:mixin}
	at net.minecraft.entity.LivingEntity.func_213340_A(LivingEntity.java:755) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:apotheosis:apothshieldblock,xf:fml:apotheosis:apothpotiondmg,pl:mixin:APP:armorcurve.mixins.json:ValueUpdateMixin,pl:mixin:APP:supplementaries.mixins.json:LivingEntityMixin,pl:mixin:APP:wardance.mixins.json:MixinAttackSpeed,pl:mixin:APP:spiderstpo.mixins.json:LivingEntityMixin,pl:mixin:APP:assets/shetiphiancore/shetiphiancore.mixins.json:MixinLivingEntity,pl:mixin:APP:META-INF/airhop.mixins.json:accessor.ILivingEntityAccessor,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.entity.EntityPredicate.func_221015_a(SourceFile:87) ~[?:?] {re:classloading}
	at net.minecraft.world.IEntityReader.func_217361_a(SourceFile:146) ~[?:?] {re:mixin,re:classloading}
	at net.minecraft.world.IEntityReader.func_217372_a(SourceFile:123) ~[?:?] {re:mixin,re:classloading}
	at net.minecraft.entity.ai.goal.NearestAttackableTargetGoal.func_220778_g(SourceFile:54) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.entity.ai.goal.NearestAttackableTargetGoal.func_75250_a(SourceFile:44) ~[?:?] {re:classloading,pl:accesstransformer:B}


	at net.minecraft.entity.monster.ShulkerEntity$AttackNearestGoal.func_75250_a(ShulkerEntity.java:498) ~[?:?] {re:classloading}


	at net.minecraft.entity.ai.goal.PrioritizedGoal.func_75250_a(SourceFile:22) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.entity.ai.goal.GoalSelector$$Lambda$24528/1084888379.test(Unknown Source) ~[?:?] {}
	at java.util.stream.ReferencePipeline$2$1.accept(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.stream.ReferencePipeline$2$1.accept(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.stream.ReferencePipeline$2$1.accept(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.stream.ReferencePipeline$2$1.accept(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.Iterator.forEachRemaining(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.stream.ReferencePipeline.forEach(Unknown Source) ~[?:1.8.0_261] {}
	at net.minecraft.entity.ai.goal.GoalSelector.func_75774_a(SourceFile:77) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.entity.MobEntity.func_70626_be(MobEntity.java:679) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:spiderstpo.mixins.json:MobEntityMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.entity.LivingEntity.func_70636_d(LivingEntity.java:2411) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:apotheosis:apothshieldblock,xf:fml:apotheosis:apothpotiondmg,pl:mixin:APP:armorcurve.mixins.json:ValueUpdateMixin,pl:mixin:APP:supplementaries.mixins.json:LivingEntityMixin,pl:mixin:APP:wardance.mixins.json:MixinAttackSpeed,pl:mixin:APP:spiderstpo.mixins.json:LivingEntityMixin,pl:mixin:APP:assets/shetiphiancore/shetiphiancore.mixins.json:MixinLivingEntity,pl:mixin:APP:META-INF/airhop.mixins.json:accessor.ILivingEntityAccessor,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.entity.MobEntity.func_70636_d(MobEntity.java:488) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:spiderstpo.mixins.json:MobEntityMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.entity.monster.ShulkerEntity.func_70636_d(ShulkerEntity.java:311) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.entity.LivingEntity.func_70071_h_(LivingEntity.java:2158) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:apotheosis:apothshieldblock,xf:fml:apotheosis:apothpotiondmg,pl:mixin:APP:armorcurve.mixins.json:ValueUpdateMixin,pl:mixin:APP:supplementaries.mixins.json:LivingEntityMixin,pl:mixin:APP:wardance.mixins.json:MixinAttackSpeed,pl:mixin:APP:spiderstpo.mixins.json:LivingEntityMixin,pl:mixin:APP:assets/shetiphiancore/shetiphiancore.mixins.json:MixinLivingEntity,pl:mixin:APP:META-INF/airhop.mixins.json:accessor.ILivingEntityAccessor,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.entity.MobEntity.func_70071_h_(MobEntity.java:300) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:spiderstpo.mixins.json:MobEntityMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.entity.monster.ShulkerEntity.func_70071_h_(ShulkerEntity.java:153) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.world.server.ServerWorld.func_217479_a(ServerWorld.java:611) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:wardance.mixins.json:MixinSoundAlert,pl:mixin:APP:quark.mixins.json:ServerWorldMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.world.server.ServerWorld$$Lambda$24505/867328280.accept(Unknown Source) ~[?:?] {}
	at net.minecraft.world.World.func_217390_a(World.java:554) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:observerlib:coremodmethod,pl:mixin:APP:sereneseasons.mixins.json:MixinWorld,pl:mixin:APP:blockshifter.mixins.json:WorldMixin,pl:mixin:APP:brazier-common.mixins.json:WorldMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.world.server.ServerWorld.func_72835_b(ServerWorld.java:404) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:wardance.mixins.json:MixinSoundAlert,pl:mixin:APP:quark.mixins.json:ServerWorldMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:851) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterendforge.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixin.terraforged.json:common.MixinMinecraftServer,pl:mixin:APP:enhancedcelestials.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:291) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:787) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterendforge.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixin.terraforged.json:common.MixinMinecraftServer,pl:mixin:APP:enhancedcelestials.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterendforge.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixin.terraforged.json:common.MixinMinecraftServer,pl:mixin:APP:enhancedcelestials.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:232) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterendforge.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixin.terraforged.json:common.MixinMinecraftServer,pl:mixin:APP:enhancedcelestials.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer$$Lambda$22589/1819529090.run(Unknown Source) ~[?:?] {}
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_261] {}
```

**Comment by No3371** [comment]
I've confirmed that It does not always happens, so this is very likely a world loading issue.

**Comment by Jackiecrazy** [comment]
Curious, I've never seen this one. I'll add a mob blacklist for visibility.

**Comment by No3371** [comment]
It shouldn't matter which mob it is though?
I've encountered a lot of world/chunk related issue with a multiple mods, basically everything fail if the code try to access a world/chunk that is not completely loaded... I'm not sure if Minecraft/Forge provides a way to check if a chunk can be safely accessed.

**Comment by Jackiecrazy** [comment]
Ah. I see. Fixed.

**Comment by Jackiecrazy** [comment]
This should be fixed in 0.5.11.
